File: 招商银行信用卡2018春季招聘研发第一批/第一题.java
import java.util.*;

public class Main {

	public Scanner cin = new Scanner(System.in);

	Main() {
		while (cin.hasNext()) {
			String str = cin.next();
			int ans = str.length() - 1;
			for (int i = 0; ans > 0; ans--) {
				for (i = 0; str.length() % ans == 0 && i < str.length()
						&& str.charAt(i) == str.charAt(i % ans); i++) {
				}
				if (i == str.length())
					break;
			}
			System.out.println(ans != 0 ? str.substring(0, ans) : "false");
		}
	}

	public static void main(String[] args) {
		new Main();
	}
}

File: Interview Code/吉比特2018春招技术类笔试试卷编程题/第一题.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    for (int n1, n2; cin >> n1 >> n2; ) {
        int ans = 0;
        unsigned int n_1 = n1, n_2 = n2;
        for (int i = 0; i < 32; i++) {
            ans += (n_1 & 1) ^ (n_2 & 1);
            n_1 >>= 1;
            n_2 >>= 1;
        }
        cout << ans << endl;
    }
    return 0;
}
